dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -57,22 +58,25 @@
       CharMatcher.inRange('a', 'z').or(CharMatcher.inRange('0', '9'));
 
   static {
-    setDetectedOsSystemProperties();
+    detectOsProperties().forEach(System::setProperty);
   }
 
   // caching cuts time by about a factor of 4.
   private static final Map<String, DependencyNode> cacheWithProvidedScope = new HashMap<>();
   private static final Map<String, DependencyNode> cacheWithoutProvidedScope = new HashMap<>();
 
-  private static void setDetectedOsSystemProperties() {
+  public static ImmutableMap<String, String> detectOsProperties() {
```

Comment:
reusing existing logic.

I tried calling `setDetectedOsSystemProperties` at the beginning of `LinkageCheckerRule.execute`. But it was too late to inject the properties into the RepositorySystemSession.